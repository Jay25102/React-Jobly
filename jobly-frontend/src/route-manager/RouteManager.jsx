import React, { useContext } from "react";
import { Routes, Route } from 'react-router-dom';
import UserContext from "../forms/UserContext";

import Homepage from "../Homepage";
import CompanyList from "../companies/CompanyList";
import CompanyDetail from "../companies/CompanyDetail";
import JobsList from "../jobs/JobsList";
import LoginForm from "../forms/LoginForm";
import SignupForm from "../forms/SignupForm";
import EditProfileForm from "../forms/EditProfileForm";

/**
 * Site-wide routes. Some routes require a logged in user.
 */
function RouteManager({ login, signup }) {
    const {currentUser} = useContext(UserContext);
    console.debug("Routes", "login=", typeof login, "signup=", typeof signup);

    if (!currentUser) {
        return (
            <Routes>
                <Route exact path="/" element={<Homepage/>}/>
                <Route exact path="/login" element={<LoginForm login={login}/>}/>
                <Route exact path="/signup" element={<SignupForm signup={signup}/>}/>
                <Route path="/*" to="/login"/>
            </Routes>
        )
    }

    return (
        <Routes>
            <Route exact path="/" element={<Homepage/>}/>
            <Route exact path="/companies" element={<CompanyList/>}/>
            <Route exact path="/companies/:handle" element={<CompanyDetail/>}/>
            <Route exact path="/jobs" element={<JobsList/>}/>
            <Route exact path="/login" element={<LoginForm login={login}/>}/>
            <Route exact path="/signup" element={<SignupForm signup={signup}/>}/>
            <Route exact path="/profile" element={<EditProfileForm/>}/>
        </Routes>
    )
}

export default RouteManager;