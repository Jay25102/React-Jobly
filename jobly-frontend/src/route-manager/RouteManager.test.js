import React from "react";
import { render } from "@testing-library/react";
import RouteManager from "./RouteManager";
import { MemoryRouter, Route } from "react-router-dom";
import { UserProvider } from "../testUtils";

describe("RouteManager", function() {
    test("it renders without crashing", function() {
        render(
            <MemoryRouter>
                <UserProvider>
                    <RouteManager/>
                </UserProvider>
            </MemoryRouter>
        );
    });
    test("it matches snapshot", function() {
        const { asFragment } = render(
            <MemoryRouter>
                <UserProvider>
                    <RouteManager/>
                </UserProvider>
            </MemoryRouter>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});