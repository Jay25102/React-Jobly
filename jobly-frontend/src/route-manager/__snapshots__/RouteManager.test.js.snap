// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RouteManager it matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Jobly
    </div>
    <div>
      All the jobs in one, convenient place.
    </div>
    <h2>
      Welcome Back, testuser!
    </h2>
  </div>
</DocumentFragment>
`;
